com.hazelcast.internal.server.tcp.ProtocolNegotiationTest.verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]

master (commit 3cb8ce1fc3bb848aced6c87a30bf7b31aec16cf7 )

Failed on MAcOS ZulluJDK8: http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-MacOS-ZuluJDK8/73/consoleFull

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected java.lang.IllegalStateException to be thrown, but nothing was thrown
	at org.junit.Assert.assertThrows(Assert.java:1028)
	at org.junit.Assert.assertThrows(Assert.java:981)
	at com.hazelcast.internal.server.tcp.ProtocolNegotiationTest.verifyOnlyTheProtocolHeaderIsSent(ProtocolNegotiationTest.java:92)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

</details>

<details><summary>Standard output:</summary>

```
19:15:14,298  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [ProtocolNegotiationTest$BytesCountingServer] Thread-13350 - The server will be listening on port 57130
19:15:14,299  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [MetricsConfigHelper] verifyOnlyTheProtocolHeaderIsSent - [LOCAL] [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
19:15:14,300  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [logo] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
19:15:14,300  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [system] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
19:15:14,300  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [system] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Hazelcast Platform 5.2-SNAPSHOT (20220525 - 3cb8ce1) starting at [127.0.0.1]:5258
19:15:14,300  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [system] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Cluster name: verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]
19:15:14,300  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [system] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
19:15:14,300  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [system] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Jet is enabled
19:15:14,303  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [MetricsConfigHelper] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
19:15:14,305  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [Node] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Using TCP/IP discovery
19:15:14,305  WARN |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [CPSubsystem] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
19:15:14,306 DEBUG |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [JetServiceBackend] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Jet exceptions are not registered to the ClientExceptionFactory since the ClientExceptionFactory is not accessible.
19:15:14,306  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [JetServiceBackend] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
19:15:14,308  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [Diagnostics] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
19:15:14,308  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [LifecycleService] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] [127.0.0.1]:5258 is STARTING
19:15:14,309  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [TcpServerConnection] hz.zen_elion.IO.thread-out-0 - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:57131 and /127.0.0.1:57130
19:15:14,309  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [ProtocolNegotiationTest$BytesCountingServer] Thread-13353 - Socket accepted Socket[addr=/127.0.0.1,port=57131,localport=57130]
19:15:16,319  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [ProtocolNegotiationTest$BytesCountingServer] Thread-13353 - Bytes read: 3
19:15:16,320  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [TcpServerConnection] hz.zen_elion.IO.thread-in-0 - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Connection[id=1, /127.0.0.1:57131->/127.0.0.1:57130, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:57130, remoteUuid=null, alive=false, connectionType=MEMBER, planeIndex=-1] closed. Reason: Connection closed by the other side
19:15:16,320  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [TcpIpJoiner] hz.zen_elion.IO.thread-in-0 - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] [127.0.0.1]:57130 is added to the blacklist.
19:15:16,329  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [ClusterService] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5258 - 7f262f4b-6566-4aac-88e8-c9c6af207a12 this
]

19:15:16,329  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [JobCoordinationService] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Jet started scanning for jobs
19:15:16,329 DEBUG |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [JobCoordinationService] hz.zen_elion.cached.thread-6 - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
19:15:16,330  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [LifecycleService] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] [127.0.0.1]:5258 is STARTED
19:15:16,330  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [ProtocolNegotiationTest$BytesCountingServer] Thread-13355 - Socket accepted Socket[addr=/127.0.0.1,port=57132,localport=57130]
19:15:16,330  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [TcpServerConnection] hz.zen_elion.IO.thread-out-1 - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:57132 and /127.0.0.1:57130
19:15:16,421  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [LifecycleService] Thread-13350 - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] [127.0.0.1]:5258 is SHUTTING_DOWN
19:15:16,421  WARN |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [Node] Thread-13350 - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Terminating forcefully...
19:15:16,421  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [Node] Thread-13350 - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Shutting down connection manager...
19:15:16,421  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [TcpServerConnection] Thread-13350 - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Connection[id=2, /127.0.0.1:57132->/127.0.0.1:57130, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:57130, remoteUuid=null, alive=false, connectionType=MEMBER, planeIndex=-1] closed. Reason: TcpServer is stopping
19:15:16,422  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [TcpIpJoiner] Thread-13350 - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] [127.0.0.1]:57130 is added to the blacklist.
19:15:16,422  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [Node] Thread-13350 - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Shutting down node engine...
19:15:16,422  WARN |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [ProtocolNegotiationTest$BytesCountingServer] Thread-13355 - Reading from the socket failed
java.net.SocketException: Connection reset
	at java.net.SocketInputStream.read(SocketInputStream.java:210) ~[?:1.8.0_302]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_302]
	at java.net.SocketInputStream.read(SocketInputStream.java:224) ~[?:1.8.0_302]
	at com.hazelcast.internal.server.tcp.ProtocolNegotiationTest$BytesCountingServer.readWithTimeout(ProtocolNegotiationTest.java:184) ~[test-classes/:?]
	at com.hazelcast.internal.server.tcp.ProtocolNegotiationTest$BytesCountingServer.lambda$run$1(ProtocolNegotiationTest.java:155) ~[test-classes/:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_302]
19:15:16,423  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [NodeExtension] Thread-13350 - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Destroying node NodeExtension.
19:15:16,423  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [Node] Thread-13350 - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] Hazelcast Shutdown is completed in 2 ms.
19:15:16,423  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [LifecycleService] Thread-13350 - [127.0.0.1]:5258 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]] [5.2-SNAPSHOT] [127.0.0.1]:5258 is SHUTDOWN
19:15:16,423  WARN |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [ProtocolNegotiationTest$BytesCountingServer] Thread-13351 - The test server thrown an exception
java.net.SocketException: Socket closed
	at java.net.PlainSocketImpl.socketAccept(Native Method) ~[?:1.8.0_302]
	at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:409) ~[?:1.8.0_302]
	at java.net.ServerSocket.implAccept(ServerSocket.java:562) ~[?:1.8.0_302]
	at java.net.ServerSocket.accept(ServerSocket.java:530) ~[?:1.8.0_302]
	at com.hazelcast.internal.server.tcp.ProtocolNegotiationTest$BytesCountingServer.run(ProtocolNegotiationTest.java:147) [test-classes/:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_302]
Standard Error
THREAD DUMP FOR TEST FAILURE: "expected java.lang.IllegalStateException to be thrown, but nothing was thrown" at "verifyOnlyTheProtocolHeaderIsSent"

"Thread-13355" 
	java.lang.Thread.State: RUNNABLE, cpu=294000 nsecs, usr=156000 nsecs, blocked=0 msecs, waited=0 msecs
		at java.net.SocketInputStream.socketRead0(Native Method)
		at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
		at java.net.SocketInputStream.read(SocketInputStream.java:171)
		at java.net.SocketInputStream.read(SocketInputStream.java:141)
		at java.net.SocketInputStream.read(SocketInputStream.java:224)
		at com.hazelcast.internal.server.tcp.ProtocolNegotiationTest$BytesCountingServer.readWithTimeout(ProtocolNegotiationTest.java:184)
		at com.hazelcast.internal.server.tcp.ProtocolNegotiationTest$BytesCountingServer.lambda$run$1(ProtocolNegotiationTest.java:155)
		at com.hazelcast.internal.server.tcp.ProtocolNegotiationTest$BytesCountingServer$$Lambda$14477/451185554.run(Unknown Source)
		at java.lang.Thread.run(Thread.java:748)

"hz.zen_elion.HealthMonitor" 
	java.lang.Thread.State: TIMED_WAITING, cpu=161000 nsecs, usr=60000 nsecs, blocked=0 msecs, waited=86 msecs
		at java.lang.Thread.sleep(Native Method)
		at java.lang.Thread.sleep(Thread.java:340)
		at java.util.concurrent.TimeUnit.sleep(TimeUnit.java:386)
		at com.hazelcast.internal.diagnostics.HealthMonitor$HealthMonitorThread.run(HealthMonitor.java:163)

"zen_elion-NioNetworking-closeListenerExecutor" 
	java.lang.Thread.State: WAITING, on lock=java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject@59abcbc4, cpu=152000 nsecs, usr=67000 nsecs, blocked=0 msecs, waited=95 msecs
		at sun.misc.Unsafe.park(Native Method)
		at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
		at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2044)
		at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
		at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
		at java.lang.Thread.run(Thread.java:748)

"hz.zen_elion.cached.thread-9" 
	java.lang.Thread.State: TIMED_WAITING, on lock=java.util.concurrent.SynchronousQueue$TransferStack@576c3e81, cpu=658000 nsecs, usr=265000 nsecs, blocked=0 msecs, waited=1108 msecs
		at sun.misc.Unsafe.park(Native Method)
		at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
		at java.util.concurrent.SynchronousQueue$TransferStack.awaitFulfill(SynchronousQueue.java:460)
		at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:362)
		at java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:941)
		at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1073)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
		at java.lang.Thread.run(Thread.java:748)
		at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
		at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)

"hz.zen_elion.cached.thread-8" 
	java.lang.Thread.State: TIMED_WAITING, on lock=java.util.concurrent.SynchronousQueue$TransferStack@576c3e81, cpu=233000 nsecs, usr=93000 nsecs, blocked=0 msecs, waited=1109 msecs
		at sun.misc.Unsafe.park(Native Method)
		at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
		at java.util.concurrent.SynchronousQueue$TransferStack.awaitFulfill(SynchronousQueue.java:460)
		at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:362)
		at java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:941)
		at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1073)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
		at java.lang.Thread.run(Thread.java:748)
		at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
		at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)

"hz.zen_elion.cached.thread-7" 
	java.lang.Thread.State: TIMED_WAITING, on lock=java.util.concurrent.SynchronousQueue$TransferStack@576c3e81, cpu=680000 nsecs, usr=326000 nsecs, blocked=0 msecs, waited=1108 msecs
		at sun.misc.Unsafe.park(Native Method)
		at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
		at java.util.concurrent.SynchronousQueue$TransferStack.awaitFulfill(SynchronousQueue.java:460)
		at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:362)
		at java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:941)
		at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1073)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
		at java.lang.Thread.run(Thread.java:748)
		at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
		at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)

"hz.zen_elion.cached.thread-6" 
	java.lang.Thread.State: TIMED_WAITING, on lock=java.util.concurrent.SynchronousQueue$TransferStack@576c3e81, cpu=352000 nsecs, usr=170000 nsecs, blocked=0 msecs, waited=1109 msecs
		at sun.misc.Unsafe.park(Native Method)
		at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
		at java.util.concurrent.SynchronousQueue$TransferStack.awaitFulfill(SynchronousQueue.java:460)
		at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:362)
		at java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:941)
		at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1073)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
		at java.lang.Thread.run(Thread.java:748)
		at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
		at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)

"hz.zen_elion.cached.thread-5" 
```

</details>

Failed on PR builder https://jenkins.hazelcast.com/job/Hazelcast-pr-builder/12856/testReport/junit/com.hazelcast.internal.server.tcp/ProtocolNegotiationTest/verifyOnlyTheProtocolHeaderIsSent_advancedNetworking_false_/

`com.hazelcast.internal.server.tcp.ProtocolNegotiationTest.verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]`


Error Message

```
expected java.lang.IllegalStateException to be thrown, but nothing was thrown

```

Stacktrace

```
java.lang.AssertionError: expected java.lang.IllegalStateException to be thrown, but nothing was thrown
	at org.junit.Assert.assertThrows(Assert.java:1028)
	at org.junit.Assert.assertThrows(Assert.java:981)
	at com.hazelcast.internal.server.tcp.ProtocolNegotiationTest.verifyOnlyTheProtocolHeaderIsSent(ProtocolNegotiationTest.java:92)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

stdout
```
Finished Running Test: verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true] in 8.032 seconds.
10:59:54,125  WARN |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:true]| - [ProtocolNegotiationTest$BytesCountingServer] Thread-457 - The test server thrown an exception
java.net.SocketException: Socket closed
	at java.net.PlainSocketImpl.socketAccept(Native Method) ~[?:1.8.0_311]
	at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:535) ~[?:1.8.0_311]
	at java.net.ServerSocket.implAccept(ServerSocket.java:545) ~[?:1.8.0_311]
	at java.net.ServerSocket.accept(ServerSocket.java:513) ~[?:1.8.0_311]
	at com.hazelcast.internal.server.tcp.ProtocolNegotiationTest$BytesCountingServer.run(ProtocolNegotiationTest.java:147) ~[test-classes/:?]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_311]
Started Running Test: verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]
10:59:54,139  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [ProtocolNegotiationTest$BytesCountingServer] Thread-462 - The server will be listening on port 46875
10:59:54,142  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [MetricsConfigHelper] verifyOnlyTheProtocolHeaderIsSent - [LOCAL] [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
10:59:54,143  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [logo] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
10:59:54,143  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [system] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
10:59:54,143  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [system] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Hazelcast Platform 5.2-SNAPSHOT (20220829 - f99e83d) starting at [127.0.0.1]:5014
10:59:54,143  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [system] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Cluster name: verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]
10:59:54,143  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [system] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
10:59:54,143  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [system] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Jet is enabled
10:59:54,151  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [MetricsConfigHelper] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
10:59:54,159  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [Node] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Using TCP/IP discovery
10:59:54,159  WARN |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [CPSubsystem] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
10:59:54,163  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [JetServiceBackend] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
10:59:54,165  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [Diagnostics] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
10:59:54,165  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [LifecycleService] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] [127.0.0.1]:5014 is STARTING
10:59:54,167  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [ProtocolNegotiationTest$BytesCountingServer] Thread-465 - Socket accepted Socket[addr=/127.0.0.1,port=39861,localport=46875]
10:59:56,206  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [ProtocolNegotiationTest$BytesCountingServer] Thread-465 - Bytes read: 3
10:59:56,207  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [TcpServerConnection] hz.kind_meninsky.IO.thread-in-0 - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Connection[id=1, /0.0.0.0:39861->/127.0.0.1:46875, qualifier=null, endpoint=[127.0.0.1]:46875, remoteUuid=null, alive=false, connectionType=NONE, planeIndex=-1] closed. Reason: Connection closed by the other side
10:59:56,207  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [TcpIpJoiner] hz.kind_meninsky.IO.thread-in-0 - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] [127.0.0.1]:46875 is added to the blacklist.
10:59:56,406  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [ClusterService] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5014 - 92fc86cb-e530-4367-9113-9c774bd75fa9 this
]

10:59:56,406  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [JobCoordinationService] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Jet started scanning for jobs
10:59:56,406 DEBUG |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [JobCoordinationService] hz.kind_meninsky.cached.thread-2 - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
10:59:56,407  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [LifecycleService] verifyOnlyTheProtocolHeaderIsSent - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] [127.0.0.1]:5014 is STARTED
10:59:56,408  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [ProtocolNegotiationTest$BytesCountingServer] Thread-467 - Socket accepted Socket[addr=/127.0.0.1,port=52621,localport=46875]
10:59:56,546 DEBUG |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [JobCoordinationService] hz.kind_meninsky.cached.thread-6 - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
10:59:56,573  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [LifecycleService] Thread-462 - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] [127.0.0.1]:5014 is SHUTTING_DOWN
10:59:56,573  WARN |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [Node] Thread-462 - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Terminating forcefully...
10:59:56,573  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [Node] Thread-462 - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Shutting down connection manager...
10:59:56,575  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [TcpServerConnection] Thread-462 - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Connection[id=2, /0.0.0.0:52621->/127.0.0.1:46875, qualifier=null, endpoint=[127.0.0.1]:46875, remoteUuid=null, alive=false, connectionType=NONE, planeIndex=-1] closed. Reason: TcpServer is stopping
10:59:56,575  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [TcpIpJoiner] Thread-462 - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] [127.0.0.1]:46875 is added to the blacklist.
10:59:56,576  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [Node] Thread-462 - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Shutting down node engine...
10:59:56,579  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [NodeExtension] Thread-462 - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Destroying node NodeExtension.
10:59:56,579  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [Node] Thread-462 - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] Hazelcast Shutdown is completed in 6 ms.
10:59:56,579  INFO |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [LifecycleService] Thread-462 - [127.0.0.1]:5014 [verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]] [5.2-SNAPSHOT] [127.0.0.1]:5014 is SHUTDOWN
10:59:56,580  WARN |verifyOnlyTheProtocolHeaderIsSent[advancedNetworking:false]| - [ProtocolNegotiationTest$BytesCountingServer] Thread-463 - The test server thrown an exception
java.net.SocketException: Socket closed
	at java.net.PlainSocketImpl.socketAccept(Native Method) ~[?:1.8.0_311]
	at java.net.AbstractPlainSocketImpl.accept(AbstractPlainSocketImpl.java:535) ~[?:1.8.0_311]
	at java.net.ServerSocket.implAccept(ServerSocket.java:545) ~[?:1.8.0_311]
	at java.net.ServerSocket.accept(ServerSocket.java:513) ~[?:1.8.0_311]
	at com.hazelcast.internal.server.tcp.ProtocolNegotiationTest$BytesCountingServer.run(ProtocolNegotiationTest.java:147) ~[test-classes/:?]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_311]
```

--- a/hazelcast/src/test/java/com/hazelcast/internal/server/tcp/ProtocolNegotiationTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/server/tcp/ProtocolNegotiationTest.java
@@ -60,11 +60,12 @@ import com.hazelcast.test.annotation.QuickTest;
 @Category({ QuickTest.class })
 public class ProtocolNegotiationTest {
 
-    private final BytesCountingServer bcServer = new BytesCountingServer(createServerSocket());
+    private volatile BytesCountingServer bcServer;
     private final TestAwareInstanceFactory factory = new TestAwareInstanceFactory();
 
     @Before
     public void before() {
+        bcServer = new BytesCountingServer(createServerSocket());
         new Thread(bcServer).start();
     }
 

